------------------------------------------------------------
revno: 14145
revision-id: squid3@treenet.co.nz-20170226085009-tj3o81s5ybk8ly9s
parent: squid3@treenet.co.nz-20170226084624-5tkl3bdrqz8nlp9g
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Sun 2017-02-26 21:50:09 +1300
message:
  Fix missing CRLF on FTP timeout ABORT commands
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170226085009-tj3o81s5ybk8ly9s
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: a290f77545e17012eea87e6315a9d375bff16752
# timestamp: 2017-02-26 08:51:05 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20170226084624-\
#   5tkl3bdrqz8nlp9g
# 
# Begin patch
=== modified file 'src/clients/FtpGateway.cc'
--- src/clients/FtpGateway.cc	2017-01-01 00:16:45 +0000
+++ src/clients/FtpGateway.cc	2017-02-26 08:50:09 +0000
@@ -1775,7 +1775,7 @@
 
         // ABORT on timeouts. server may be waiting on a broken TCP link.
         if (io.xerrno == Comm::TIMEOUT)
-            writeCommand("ABOR");
+            writeCommand("ABOR\r\n");
 
         // try another connection attempt with some other method
         ftpSendPassive(this);

